File: 1-TwoSum/1-TwoSum.py
# Last updated: 7/9/2026, 3:12:06 PM
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_to_index = {}    
        for index, num in enumerate(nums):
            complement = target - num
            if complement in num_to_index:
                return [num_to_index[complement],index]
            num_to_index[num] = index

File: 1-TwoSum/1-TwoSum.java
// Last updated: 7/9/2026, 3:12:05 PM
class Solution {
    public int[] twoSum(int[] nums, int target) {
        // int[] arr = new int[2];
        Map<Integer, Integer> map = new HashMap<>();
        for (int i=0; i<nums.length; i++){
            int x = target - nums[i];
            if (map.containsKey(nums[i])){
                // arr[0] = i;
                // arr[1] = map.get(nums[i]);
                return new int[]{i, map.get(nums[i])};
            }else{
                map.put(x, i);
            }
        } 
        return new int[]{-1, -1};
    }
}